tools/UpdateDotnetRuntime.ps1

```diff
@@ -0,0 +1,130 @@
+param (
+)
+
+function Update-GlobalJson([string] $Version) {
+    $psGlobalJsonPath = Resolve-Path "$PSScriptRoot/../global.json"
+    $psGlobalJson = Get-Content -Path $psGlobalJsonPath -Raw | ConvertFrom-Json
+    $psGlobalJson.sdk.version = $Version
+    $psGlobalJson | ConvertTo-Json | Out-File -FilePath $psGlobalJsonPath -Force
+}
+
+function Update-PackageVersion {
+
+    class PkgVer {
+        [string] $Name
+        [string] $Version
+        [string] $NewVersion
+        [string] $Path
+
+        PkgVer($n, $v, $nv, $p) {
+            $this.Name = $n
+            $this.Version = $v
+            $this.NewVersion = $nv
+            $this.Path = $p
+        }
+    }
+
+    $skipModules = @(
+        "NJsonSchema"
+        "Markdig.Signed"
+        "PowerShellHelpFiles"
+        "Newtonsoft.Json"
+        "Microsoft.ApplicationInsights"
+        "Microsoft.Management.Infrastructure"
+        "Microsoft.PowerShell.Native"
+        "Microsoft.NETCore.Windows.ApiSets"
+    )
+
+    $packages = [System.Collections.Generic.Dictionary[[string], [PkgVer]]]::new()
+
+    Get-ChildItem -Path "$PSScriptRoot/../src/" -Recurse -Filter "*.csproj" -Exclude 'PSGalleryModules.csproj' | ForEach-Object {
+        $prj = [xml] (Get-Content $_.FullName -Raw)
+        $pkgRef = $prj.Project.ItemGroup.PackageReference
+
+        foreach ($p in $pkgRef) {
+            if ($null -ne $p -and -not $skipModules.Contains($p.Include)) {
+                if (-not $packages.ContainsKey($p.Include)) {
+                    $packages.Add($p.Include, [PkgVer]::new($p.Include, $p.Version, $null, $_.FullName))
+                }
+            }
+        }
+    }
+
+    $versionPattern = (Get-Content "$PSScriptRoot/../DotnetRuntimeMetadata.json" | ConvertFrom-Json).sdk.packageVersionPattern
+
+    $packages.GetEnumerator() | ForEach-Object {
+        $pkgs = Find-Package -Name $_.Key -AllVersions -AllowPreReleaseVersions -Source 'dotnet5'
+
+        $version = $_.Value.Version
+
+        foreach ($p in $pkgs) {
+            if ($p.Version -like "$versionPattern*") {
+                if ([System.Management.Automation.SemanticVersion] ($version) -lt [System.Management.Automation.SemanticVersion] ($p.Version)) {
+                    $_.Value.NewVersion = $p.Version
+                    break
+                }
+            }
+        }
+    }
+
+    $pkgsByPath = $packages.Values | Group-Object -Property Path
+
+    $pkgsByPath | ForEach-Object {
+        Update-CsprojFile -Path $_.Name -Values $_.Group
+    }
+}
+
+function Update-CsprojFile([string] $path, $values) {
+    $fileContent = Get-Content $path -raw
+    $updated = $false
+
+    foreach ($v in $values) {
+        if ($v.NewVersion) {
+            $stringToReplace = "<PackageReference Include=`"$($v.Name)`" Version=`"$($v.Version)`" />"
+            $newString = "<PackageReference Include=`"$($v.Name)`" Version=`"$($v.NewVersion)`" />"
+
+            $fileContent = $fileContent -replace $stringToReplace, $newString
```

Comment:
This is doing a regex replace... maybe it's safer to using <string>.replace()

Reply:
Regex replace is turning out to be a bit complicated. Keeping it as is. The intention is to create a PR after the script does the changes. So there will be a code review as a safety net.